### PR TITLE
Removed the ability to run the -O and -t command line option together (#5516)

### PR DESCRIPTION
**Pull request by MagnusOffermanns**
# Fix for Issue 5516
If `-t` and `-O` are used together scrapy now throws a `usageError`. I also created a error message including examples.

Furthermore, I added an extended better deprecation warning in case `-t` and `-o`
are used together including two examples.

### CODE REVIEW

**Comment by MagnusOffermanns** [comment]
I have two inputs that I would like to get feedback on, as I do not really know how to handle the two cases. I would be very glad for your feedback.

1. The warnings and error messages I created are all in one line similar to the already existing ones. Would it not make sense to make them a little bit more readable by giving them new lines? Is it maybe wanted that they only have one line in order to display them better?
2. When the combination of `-t` and `-O` throws an error would it not also make sense to throw an error when `-t` and `-o` are used? Therefore making the `-t` option obsolete? How long do we have to keep `-t` till we can remove it from the source code?

I thank you for answering my questions and hope I could resolve this issue ok 🚀

**Comment by MagnusOffermanns** [comment]
I also extended the documentation for `crawl` and the help message. Maybe I also closed issue #5530.


**With one caviat:**
I was not able to make the `-a` command run. Therefore I did not add an example there. If someone gives me a tip how to use the `-a` option I will quickly add it as well.

Anyway, thank you for your help and feedback.

File: docs/topics/commands.rst [hunk 1]
```diff
@@ -271,11 +271,32 @@ crawl
...
+Supported options:
+
+* ``-h, --help``: show a help message and exit
+
+* ``-a NAME=VALUE``: set a spider argument (may be repeated)
+
+* ``--output FILE`` or ``-o FILE``: append scraped items to the end of FILE (use - for stdout), to define format set a colon at the end of the output URI (i.e. ``-o FILE:FORMAT``)
+
+* ``--overwrite-output FILE`` or ``-O FILE``: dump scraped items into FILE, overwriting any existing file, to define format set a colon at the end of the output URI (i.e. ``-O FILE:FORMAT``)
+
```
Comment (wRAR) [inline docs/topics/commands.rst:283]: this extra newline seems unneeded

File: docs/topics/commands.rst [hunk 2]
```diff
@@ -271,11 +271,32 @@ crawl
...
     $ scrapy crawl myspider
     [ ... myspider starts crawling ... ]
 
+    $scrapy -o myfile:csv myspider
+    [ ... myspider starts crawling and appends the result to the file myfile in csv format ... ]
+
+    $scrapy -O myfile:json myspider
+    [ ... myspider starts crawling and saves the result in myfile in json format overwriting the original content... ]
+
+    $scrapy -o myfile - t csv myspider
```
Comment (wRAR) [inline docs/topics/commands.rst:298]: extra space in `- t`

File: scrapy/utils/conf.py [hunk 3]
```diff
@@ -155,16 +155,29 @@ def check_valid_format(output_format):
             raise UsageError(
                 "Please use only one of -o/--output and -O/--overwrite-output"
             )
+        if output_format:
+            raise UsageError(
+                f"-t/--output-format is a deprecated command line option"
+                f" and does not work in combination with -O/--overwrite-output."
+                f" To specifiy a format please specify it after a colon at the end of the"
```
Comment (wRAR) [inline scrapy/utils/conf.py:162]: a typo in "specifiy"

File: scrapy/utils/conf.py [hunk 4]
```diff
@@ -155,16 +155,29 @@ def check_valid_format(output_format):
...
 
     if output_format:
         if len(output) == 1:
             check_valid_format(output_format)
             message = (
-                'The -t command line option is deprecated in favor of '
-                'specifying the output format within the output URI. See the '
-                'documentation of the -o and -O options for more information.'
+                "The -t/--output-format command line option is deprecated in favor of "
+                "pecifying the output format within the output URI using the -o/--output or the"
```
Comment (wRAR) [inline scrapy/utils/conf.py:175]: a typo in "pecifying"

**Review by wRAR — CHANGES_REQUESTED** [review]
Looks good but there are some typos, can you please fix those?

File: docs/topics/commands.rst (hunk 1 above)
Comment (MagnusOffermanns) [reply docs/topics/commands.rst:283]: Thank you for looking over my code and spotting my spelling errors. I have corrected the typos now.

**Review by wRAR — APPROVED** [review]

**Review by Gallaecio — APPROVED** [review]

**Pull request merged**